src/serialize.rs

```diff
@@ -158,12 +159,13 @@ pub fn node_from_stream(allocator: &mut Allocator, f: &mut Cursor<&[u8]>) -> io:
                     values.push(allocator.new_atom(&b)?);
                 } else {
                     let blob_size = decode_size(f, b[0])?;
-                    if (f.get_ref().len() as u64) < blob_size {
+                    let pos = f.position() as usize;
+                    if f.get_ref().len() < pos + blob_size as usize {
                         return Err(bad_encoding());
                     }
-                    let mut blob: Vec<u8> = vec![0; blob_size as usize];
-                    f.read_exact(&mut blob)?;
-                    values.push(allocator.new_atom(&blob)?);
+                    let blob = &f.get_ref()[pos..(pos + blob_size as usize)];
+                    f.seek(SeekFrom::Current(blob_size as i64))?;
+                    values.push(allocator.new_atom(blob)?);
```

Comment:
nice, makes sense to hold just the slice ref since it's going to be copied into allocator via new_atom anyway.